### PR TITLE
Patch react-virtualized to fix react 18 scroll lag

### PR DESCRIPTION
**Pull request by abaevbog**
React 18 introduced automatic batching which tries to avoid unnecessary re-rendering (https://react.dev/blog/2022/03/08/react-18-upgrade-guide#automatic-batching). In some components based on `react-virtualized` (e.g. tag selector), it leads to visible lagginess on scroll via keypress or mouse wheel (not trackpad). This patch wraps setState of the scroll handler of `react-virtualized` with `ReactDOM.flushSync`, which opts out of automatic batching.

Fixes: #4368

Make sure to run `npm run clean` before `./app/scripts/build_and_run -rd` to have `babel-worker.js` rerun.

### CODE REVIEW

**Comment by dstillman** [comment]
Aren't we fine with this in other places, though? Should we not just override this in the components where it's actually a problem?

Or if this is the best option, we should check for the pattern we're replacing first and throw if it's not present (like we do in fetch_xulrunner) — otherwise this could regress silently on a React update.

**Comment by dstillman** [comment]
Oh, yeah, that's fair.

OK, then I would just have it throw if the search pattern doesn't exist.

**Comment by abaevbog** [comment]
Cool, how about this? Correct me if the error sounds a bit off - I tried to be brief but informative with the exact line that we're trying to find.

File: js-build/babel-worker.js [hunk 1]
```diff
@@ -46,6 +46,16 @@ async function babelWorker(ev) {
...
 				.replace('document.body.removeChild(scrollDiv)', 'document.documentElement.removeChild(scrollDiv)');
+			// React 18: wrap setState in onScroll handler with ReactDOM.flushSync to avoid
+			// automatic batching https://react.dev/blog/2022/03/08/react-18-upgrade-guide#automatic-batching
+			// which causes less frequent re-rendering and lagginess on scroll of components such as tag selector
+			let onScrollSetStateChunkRegex = /(_this\.state\.isScrolling\s*\|\|\s*isScrollingChange\(!0\),\s*)(_this\.setState\s*\(\s*\{[^}]*\}\s*\))/;
+			if (!onScrollSetStateChunkRegex.test(transformed)) {
+				throw new Error(`"_this.state.isScrolling || isScrollingChange(!0), _this.setState({" not found in react-virtualized`);
+			}
+			transformed = transformed.replace(onScrollSetStateChunkRegex, (_, p1, p2) => {
+				return `${p1}ReactDOM.flushSync(() => ${p2.trim()})`;
```
Comment (dstillman) [inline js-build/babel-worker.js:57]: Already merged, but why is there a `trim()` here? p2 will always start with `_` and end with `)`, no?

Comment (abaevbog) [reply js-build/babel-worker.js:57]: You're right. When testing this out, I got thrown off by an extra new line `console.log`-ed during the build stage, so I added `.trim()` thinking it was needed for the `;` to fall at the right place. But, yeah, apologies, `.trim()` does not actually have any effect here.

Comment (abaevbog) [reply js-build/babel-worker.js:57]: I added #4379 to clean it up

**Pull request merged**